cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/NodeConversion.java

```diff
@@ -61,33 +70,122 @@ public boolean valid() {
+    }
+
+    public void setVoltageAngleNodeBreaker() {
+        if (!context.nodeBreaker()) {
+            return;
+        }
+        VoltageLevel vl = voltageLevel();
+        Objects.requireNonNull(vl);
+        VoltageLevel.NodeBreakerView topo = vl.getNodeBreakerView();
+        String connectivityNode = id;
+        int iidmNode = context.nodeMapping().iidmNodeForConnectivityNode(connectivityNode, vl);
+        // To obtain a bus for which we want to set voltage:
+        // If there no Terminal at this IIDM node,
+        // then find from it the first connected node with a Terminal
+        Terminal t = topo.getTerminal(iidmNode);
+        if (t == null) {
+            t = new FirstTerminalTraverser(topo, iidmNode).firstTerminal();
+        }
+        if (t == null) {
+            LOG.error("Can't find a Terminal to obtain a Bus to set Voltage, Angle. ConnectivityNode {}", id);
+            return;
+        }
+        setVoltageAngle(t.getBusView().getBus());
     }
 
     private VoltageLevel voltageLevel() {
-        String cgmesId = p.getId(CgmesNames.VOLTAGE_LEVEL);
+        String containerId = p.getId("ConnectivityNodeContainer");
+        String cgmesId = context.cgmes().container(containerId).voltageLevel();
         String iidmId = context.namingStrategy().getId(CgmesNames.VOLTAGE_LEVEL, cgmesId);
         return iidmId != null ? context.network().getVoltageLevel(iidmId) : null;
     }
 
-    private void createBus(VoltageLevel voltageLevel) {
+    private void newNode(VoltageLevel vl) {
+        VoltageLevel.NodeBreakerView nbv = vl.getNodeBreakerView();
+        String connectivityNode = id;
+        int iidmNode = context.nodeMapping().iidmNodeForConnectivityNode(connectivityNode, vl);
+
+        // Busbar sections are created for every connectivity node to be
+        // able to easily check the topology calculated by IIDM
+        // against the topology present in the CGMES model
+        if (context.config().createBusbarSectionForEveryConnectivityNode()) {
+            BusbarSection bus = nbv.newBusbarSection()
+                    .setId(context.namingStrategy().getId("Bus", id))
+                    .setName(context.namingStrategy().getName("Bus", name))
+                    .setNode(iidmNode)
+                    .add();
+            LOG.debug("    BusbarSection added at node {} : {} {} : {}", iidmNode, id, name, bus);
+        }
+    }
+
+    private void newBus(VoltageLevel voltageLevel) {
         Bus bus = voltageLevel.getBusBreakerView().newBus()
                 .setId(context.namingStrategy().getId("Bus", id))
                 .setName(context.namingStrategy().getName("Bus", name))
                 .add();
-        double v = p.asDouble("v");
-        double angle = p.asDouble("angle");
+        setVoltageAngle(bus);
+    }
+
+    private void setVoltageAngle(Bus bus) {
+        double v = p.asDouble(CgmesNames.VOLTAGE);
+        double angle = p.asDouble(CgmesNames.ANGLE);
         if (valid(v, angle)) {
```

Comment:
check that ```bus``` is not null here

Reply:
This bug as been detected on a Slovenian file described in node/breaker.

Reply:
In some configurations BusView buses (calculated buses) are not created (connectivity nodes with a single branch or transformer end, for example). If no BusView Bus is found, an attempt to obtain a configured bus (BusBreakerView level) is made.

If no Bus can be found the voltage and angle values are ignored (with an error logged).